### PR TITLE
Update cargo dependencies

### PR DESCRIPTION
**Pull request by Artturin**
and make sure it still builds

ran
```
cargo upgrade
cargo update
```

@Sciencentistguy

### CODE REVIEW

**Comment by Artturin** [comment]
Darwin failure is due to a flaky curl test

**Comment by Artturin** [comment]
tested with this overlay

```nix
(self: super: {
  rnix-lsp = super.rnix-lsp.overrideAttrs (oldAttrs: rec {
    version = "master";
                                                                          
    src = super.fetchFromGitHub {
      owner = "nix-community";
      repo = "rnix-lsp";
      rev = "fce066418709e9fea62fc5ce1ef1a3d4a526a995";
      sha256 = "sha256-x1xg2rokBPpowXzEPkEGo56N/TZv0yYGXAqn8UK3CU0=";
    };
                                                                          
    cargoDeps = oldAttrs.cargoDeps.overrideAttrs (lib.const {
      name = "rnix-lsp-vendor.tar.gz";
      inherit src;
      outputHash = "sha256-70MfuzPMgaUzCfbS+lZkexr8QQOm9EJDryKSSf2rxVw=";
    });
  });
})
```

File: Cargo.toml [hunk 1]
```diff
@@ -11,23 +11,23 @@ name = "rnix-lsp"
...
-regex = "1"
+gc = { version = "0.4.1", features = ["derive"] }
+lazy_static = "1.4.0"
+libc = "0.2.126"
+log = "0.4.17"
+lsp-server = "0.6.0"
+lsp-types = { version = "0.93.0", features = ["proposed"] }
+regex = "1.5.6"
 rnix = "0.10.2"
-rowan = "0.12.6"
```
Comment (Ma27) [inline Cargo.toml:24]: Is `rowan` even used here? If yes, I'd be surprised that this works fine considering that `rnix-parser` is still using 0.12.

Comment (Artturin) [reply Cargo.toml:24]: According to github search its not used here and was added in https://github.com/nix-community/rnix-lsp/commit/bcf8963ce865c4f94c7e01e63861b87e48b71188

Comment (Ma27) [reply Cargo.toml:24]: OK, I'd be in favor of removing it then. Rest LGTM.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**